.drone.yml

```diff
@@ -42,68 +75,74 @@ pipeline:
       - CGO_ENABLED=0
       - GOARM=7
     commands:
-      - go build -v -ldflags "-X main.build=${DRONE_BUILD_NUMBER}" -a -o release/linux/arm/drone-downstream
-
-  # build_windows_amd64:
-  #   image: golang:1.9-nanoserver
-  #   pull: true
-  #   group: build
-  #   environment:
-  #     - GOOS=windows
-  #     - GOARCH=amd64
-  #     - CGO_ENABLED=0
-  #   commands:
-  #     - go build -v -ldflags "-X main.build=${DRONE_BUILD_NUMBER}" -a -o release/windows/amd64/drone-downstream
+      - |
+        if test "${DRONE_TAG}" = ""; then
+          go build -v -ldflags "-X main.build=${DRONE_BUILD_NUMBER}" -a -o release/linux/arm/drone-downstream
+        else
+          go build -v -ldflags "-X main.version=${DRONE_TAG##v} -X main.build=${DRONE_BUILD_NUMBER}" -a -o release/linux/arm/drone-downstream
+        fi
 
   publish_linux_amd64:
-    image: plugins/docker:17.05
+    image: plugins/docker:17.12
     pull: true
-    repo: plugins/downstream
-    tags: [ latest, 1.0.0, 1.0, 1 ]
     secrets: [ docker_username, docker_password ]
+    group: docker
+    repo: plugins/downstream
+    auto_tag: true
+    auto_tag_suffix: linux-amd64
     dockerfile: Dockerfile
     when:
-      branch: master
-      event: push
+      event: [ push, tag ]
```

Comment:
Are these necessary?

Reply:
Yes, it should not be built and published on pull requests.

Reply:
AFAIK the `auto_tag` logic does that for you.

http://plugins.drone.io/drone-plugins/drone-docker/#autotag

> When the event type is push and the target branch is your default branch (e.g. master) the plugin will automatically tag the image as latest. All other event types and branches are ignored.

Reply:
But this will always execute the step and you have to go into the build output to see that it have not done anything.